File: lista java/at14.java
public class Configuracao {

    private static Configuracao instance;  
    private String valor;  

  
    private Configuracao() {}

    
    public static Configuracao getInstance() {
        if (instance == null) {
            instance = new Configuracao();
        }
        return instance;
    }

    public String getValor() {
        return valor;
    }

    public void setValor(String valor) {
        this.valor = valor;
    }

    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstance();
        Configuracao config2 = Configuracao.getInstance();

        config1.setValor("Configuração 1");
        System.out.println("Config1: " + config1.getValor());
        System.out.println("Config2: " + config2.getValor());  

        System.out.println("São a mesma instância? " + (config1 == config2));
    }
}

File: lista Go/at14.go
package main

import (
    "fmt"
    "sync"
)

type Configuracao struct {
    Valor string
}

var instance *Configuracao
var once sync.Once  


func GetInstance() *Configuracao {
    once.Do(func() {
        instance = &Configuracao{}
    })
    return instance
}

func main() {
    config1 := GetInstance()
    config2 := GetInstance()

    config1.Valor = "Configuração 1"
    fmt.Println("Config1:", config1.Valor)
    fmt.Println("Config2:", config2.Valor)  

    fmt.Println("São a mesma instância?", config1 == config2)
}
